test/sm4_internal_test.c

```diff
@@ -75,12 +75,164 @@ static int test_sm4_ecb(void)
 
     return 1;
 }
+
+static int test_sm4_cipher(const EVP_CIPHER *cipher,
+                           const unsigned char *key,
+                           const unsigned char *iv,
+                           const unsigned char *input,
+                           const unsigned char *expected,
+                           unsigned int len)
+{
+    EVP_CIPHER_CTX *ctx = NULL;
+    unsigned char ctext[1024];
```

Comment:
Nitpicking: `1024` seems too much adequate for those test cases...

Reply:
That's right, only 80 bytes are used at most.